src/realm/alloc.hpp

```diff
@@ -158,10 +158,36 @@ class Allocator {
     // The ref translation splits the full ref-space (both below and above baseline)
     // into equal chunks.
     struct RefTranslation {
-        char* mapping_addr;
+        char* mapping_addr = nullptr;
+        std::atomic<size_t> lowest_possible_xover_offset = 0;
+
+        // member 'xover_mapping_addr' is used for memory synchronization of the fields
```

Comment:
I think it also establishes an ordering for `lowest_possible_xover_offest` such that it will never be updated after this goes non-null.